src/Materials/effectRenderer.ts

```diff
@@ -0,0 +1,126 @@
+    }
+
+    /**
+     * renders one or more effects to a specified texture
+     * @param effectWrappers list of effects to renderer
+     * @param outputTexture texture to draw to, if null it will render to the screen
+     */
+    render(effectWrappers: Array<EffectWrapper> | EffectWrapper, outputTexture: Nullable<Texture> = null) {
+        if (!Array.isArray(effectWrappers)) {
+            effectWrappers = [effectWrappers];
+        }
+        effectWrappers.forEach((effectWrapper, i) => {
+            var renderTo = outputTexture;
+            if ((effectWrappers as Array<EffectWrapper>).length > 1 && i != (effectWrappers as Array<EffectWrapper>).length - 1) {
+                if (!this._tmpScreenBuffer) {
+                    var internalTexture = this.engine.createRenderTargetTexture(
+                        {
+                            width: Math.floor(this.engine.getRenderWidth(true)),
+                            height: Math.floor(this.engine.getRenderHeight(true)),
+                        },
+                        {
+                            generateDepthBuffer: false,
+                            generateStencilBuffer: false,
+                            generateMipMaps: false,
+                            samplingMode: Constants.TEXTURE_NEAREST_NEAREST,
+                        },
+                    );
+                    this._tmpScreenBuffer = new Texture("", null);
+                    this._tmpScreenBuffer._texture = internalTexture;
+                }
+                renderTo = this._tmpScreenBuffer;
+            }else {
+                renderTo = outputTexture;
+            }
+
+            // for any next effect make it's input the output of the previous effect
+            if (i !== 0) {
+                effectWrapper.effect.onBindObservable.addOnce(() => {
+                    effectWrapper.effect.setTexture("textureSampler", this._tmpScreenBuffer);
+                });
+            }
+
+            // Reset state
+            this.engine.setViewport(new Viewport(0, 0, 1, 1));
+            this.engine.enableEffect(effectWrapper.effect);
+
+            // Bind buffers
+            if (renderTo) {
+                this.engine.bindFramebuffer(renderTo.getInternalTexture()!);
+            }
+            this.engine.bindBuffers(this._vertexBuffers, this._indexBuffer, effectWrapper.effect);
+            effectWrapper.onApplyObservable.notifyObservers({});
+
+            // Render
+            this.engine.drawElementsType(Constants.MATERIAL_TriangleFillMode, 0, 6);
+            if (renderTo) {
+                this.engine.unBindFramebuffer(renderTo.getInternalTexture()!);
+            }
+        });
+    }
+
+    /**
+     * Disposes of the effect renderer
+     */
+    dispose() {
+        if (this._tmpScreenBuffer) {
+            this._tmpScreenBuffer.dispose();
+        }
+    }
+}
+
+/**
+ * Wraps an effect to be used for rendering
+ */
+export class EffectWrapper {

```

Comment:
Based on the comment right after counter is not needed and we also need a dispose to be able to clean the effect ?